src/test/java/com/commercetools/sync/sdk2/commons/utils/CustomUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,1041 @@
+import com.commercetools.api.models.common.AssetBuilder;
+import com.commercetools.api.models.common.AssetDraft;
+import com.commercetools.api.models.common.AssetDraftBuilder;
+import com.commercetools.api.models.common.Price;
+import com.commercetools.api.models.common.PriceBuilder;
+import com.commercetools.api.models.product.ProductDraft;
+import com.commercetools.api.models.product.ProductProjection;
+import com.commercetools.api.models.product.ProductSetAssetCustomFieldAction;
+import com.commercetools.api.models.product.ProductSetAssetCustomFieldActionBuilder;
+import com.commercetools.api.models.product.ProductSetAssetCustomTypeAction;
+import com.commercetools.api.models.product.ProductSetProductPriceCustomFieldAction;
+import com.commercetools.api.models.product.ProductSetProductPriceCustomTypeAction;
+import com.commercetools.api.models.product.ProductUpdateAction;
+import com.commercetools.api.models.type.CustomFields;
+import com.commercetools.api.models.type.CustomFieldsBuilder;
+import com.commercetools.api.models.type.CustomFieldsDraft;
+import com.commercetools.api.models.type.CustomFieldsDraftBuilder;
+import com.commercetools.api.models.type.FieldContainer;
+import com.commercetools.api.models.type.FieldContainerBuilder;
+import com.commercetools.api.models.type.ResourceTypeId;
+import com.commercetools.api.models.type.TypeReference;
+import com.commercetools.api.models.type.TypeReferenceBuilder;
+import com.commercetools.api.models.type.TypeResourceIdentifier;
+import com.commercetools.api.models.type.TypeResourceIdentifierBuilder;
+import com.commercetools.sync.sdk2.categories.CategorySyncOptions;
+import com.commercetools.sync.sdk2.categories.CategorySyncOptionsBuilder;
+import com.commercetools.sync.sdk2.categories.helpers.CategoryCustomActionBuilder;
+import com.commercetools.sync.sdk2.commons.exceptions.BuildUpdateActionException;
+import com.commercetools.sync.sdk2.commons.exceptions.SyncException;
+import com.commercetools.sync.sdk2.commons.models.AssetCustomTypeAdapter;
+import com.commercetools.sync.sdk2.commons.models.AssetDraftCustomTypeAdapter;
+import com.commercetools.sync.sdk2.products.ProductSyncOptions;
+import com.commercetools.sync.sdk2.products.ProductSyncOptionsBuilder;
+import com.commercetools.sync.sdk2.products.helpers.AssetCustomActionBuilder;
+import com.commercetools.sync.sdk2.products.helpers.PriceCustomActionBuilder;
+import com.commercetools.sync.sdk2.products.models.PriceCustomTypeAdapter;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.junit.jupiter.api.Test;
+
+class CustomUpdateActionUtilsTest {
+  private static final ProjectApiRoot CTP_CLIENT = mock(ProjectApiRoot.class);
+  private static final CategorySyncOptions CATEGORY_SYNC_OPTIONS =
+      CategorySyncOptionsBuilder.of(CTP_CLIENT).build();
+  final ProductDraft maiNewResource = mock(ProductDraft.class);
```

Comment:
The variable name is weird. What does it mean?

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/967/commits/48f0211b3fe83f72a6f5156ca1b4cc4fd0245480